.github/workflows/draft_release.yml

```diff
@@ -0,0 +1,72 @@
+name: Draft Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Release date YYYY.MM.DD Also used to generate the tag name.'
+        required: true
+
+jobs:
+  draft_release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Create Release Draft
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        shell: bash
+        run: |
+          # Format current date as YYYY.MM.DD
+
+          if [[ "${{ github.event_name}}" == 'workflow_dispatch' ]]; then
+            tag="${{ github.event.inputs.tag }}"
+          else
+            tag=$(date +'%Y.%m.%d')
+          fi
+
+          #Validate the tag is formatted correctly YYYY.MM.DD or YYYY.MM.DD-rcX
+          # where rcX is a whole number greater than zero
+          if [[ ! $tag =~ ^[0-9]{4}\.[0-9]{2}\.[0-9]{2}(-rc[0-9]+)?$ ]]; then
+            echo "Invalid tag format. Must be YYYY.MM.DD or YYYY.MM.DD-rcX"
+            exit 1
+          fi
```

Comment:
What's the `YYYY.MM.DD-rcX` for?  It's new.

Reply:
Need to support cherry pick releases. 2024.09.05-23 for example. Good catch though, I accidentally included rc in the regex.. will remove.